lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -964,7 +1130,65 @@ class BuildWhileGroups : public calyx::FuncOpPartialLoweringPattern {
         initGroups.push_back(initGroupOp);
       }
 
-      getState<ComponentLoweringState>().setLoopInitGroups(whileOp, initGroups);
+      getState<ComponentLoweringState>().setWhileLoopInitGroups(whileOp,
+                                                                initGroups);
+
+      return WalkResult::advance();
+    });
+    return res;
+  }
+};
+
+/// In BuildForGroups, a register is created for the iteration argument of
+/// the for op. This register is then initialized to the lowerBound of the for
+/// loop in a group that executes the for loop.
+class BuildForGroups : public calyx::FuncOpPartialLoweringPattern {
+  using FuncOpPartialLoweringPattern::FuncOpPartialLoweringPattern;
+
+  LogicalResult
+  partiallyLowerFuncToComp(FuncOp funcOp,
+                           PatternRewriter &rewriter) const override {
+    LogicalResult res = success();
+    funcOp.walk([&](Operation *op) {
+      // Only work on ops that support the ScfForOp.
+      if (!isa<scf::ForOp>(op))
+        return WalkResult::advance();
+
+      auto scfForOp = cast<scf::ForOp>(op);
+      ScfForOp forOp(scfForOp);
+
+      getState<ComponentLoweringState>().setUniqueName(forOp.getOperation(),
+                                                       "for");
+
+      // Create a register for the InductionVar, and set that Register as the
+      // only IterReg for the For Loop
+      auto inductionVar = forOp.getOperation().getInductionVar();
+      std::string name = getState<ComponentLoweringState>()
+                             .getUniqueName(forOp.getOperation())
+                             .str() +
+                         "_induction_var";
+      auto reg =
+          createRegister(inductionVar.getLoc(), rewriter, getComponent(),
+                         inductionVar.getType().getIntOrFloatBitWidth(), name);
+      getState<ComponentLoweringState>().addForLoopIterReg(forOp, reg, 0);
+      inductionVar.replaceAllUsesWith(reg.getOut());
+
+      // Create InitGroup that sets the InductionVar to LowerBound
+      calyx::ComponentOp componentOp =
+          getState<ComponentLoweringState>().getComponentOp();
+      SmallVector<calyx::GroupOp> initGroups;
+      std::string groupName = "init_" +
```

Comment:
Similar here, can we use llvm::join [1] ?

1: https://llvm.org/doxygen/StringExtras_8h_source.html